src/ledger/InMemoryLedgerTxn.cpp

```diff
@@ -12,6 +12,66 @@
 namespace stellar
 {
 
+// Implementation of InMemoryLedgerTxn::FilteredEntryIteratorImpl
+// The LedgerTxnRoot backed LedgerTxn commit filters out non LEDGER_ENTRYs at
+// the top level in LedgerTxnRoot. This iterator imitates that behavior by
+// skipping the same entries.
+InMemoryLedgerTxn::FilteredEntryIteratorImpl::FilteredEntryIteratorImpl(
+    EntryIterator const& begin)
+    : mIter(begin)
```

Comment:
is this correct? If `begin` is an item to filter out, it would still be returned? It seems to me that we need to also advance until we reach a "good" item

Reply:
Yeah I think you're right

Reply:
You should add a test for this specific scenario. And you should generally add a few tests for the filtering to make sure it works.

Reply:
I'm still trying to figure out how to write a test case with one of these entries at the start. Using transactions doesn't work since accounts are also modified.

Reply:
Added some test cases.